### PR TITLE
Add peffect sign, Add drive sound cart property

### PR DESCRIPTION
**Pull request by Lee0701**
This PR will add some minor improvements about sound effects.

### Changes
- Added ability to play resource pack sounds using effect sign.
- Added sign type peffect. It plays sound effect only to the players riding the train that activated it. Could 
be used for sound announcements.
- Added cart property driveSound. If set to a name of sound resource, that sound effect will be played when the train is launched from starting velocity 0. Could be used for motor sounds.

### CODE REVIEW

**Comment by bergerkiller** [comment]
I do have different ideas in mind for the 'driveEffect' portion. I want to make it an attachment that you can add and set various 'pre-triggers' for. This is important for things like animations later. But considering this is so far into the future, I think your suggested changes are fine. I can later decide to make the 'driveSound' property inject an additional attachment at runtime to do this, once this exists.

Changes for resource pack sounds are fair, though the issue remains that you can't easily put long resource path names on signs.

One change I request before I pull it though: can you remove the binary files in /lib that you (accidentally?) pushed along with it?

**Comment by ThisTNTSquid** [comment]
about the problem which signs can't hold long resource path,  we could use alias, like we have a new alias config "SoundEffectAlias.yml" , then inside the file
```yaml
soundAlias:
- "sa1" : "train.mainsys.stationannounce.1"
- "sa2" : "train.mainsys.stationannounce.2"
- "crossing" : "train.railsys.warning.crossing"
```
on the sign, we just need to put the key alias

File: src/main/java/com/bergerkiller/bukkit/tc/utils/Effect.java [hunk 1]
```diff
@@ -123,6 +124,13 @@ public void play(Location location) {
                 location.getWorld().playSound(location, sound, volume, pitch);
                 continue;
             }
+            location.getWorld().playSound(location, name, volume, pitch);
```
[the remark this answers is not in the document]
Comment (bergerkiller) [reply src/main/java/com/bergerkiller/bukkit/tc/utils/Effect.java]: True, but I dont think playing custom sound names from resource packs was supported in 1.8.8 and earlier, anyway. Could wrap the code in a Common.evaluateMCVersion(">=", "1.9").

**Comment by bergerkiller** [comment]
@ThisTNTSquid honestly, why not just extend this logic to _all_ signs and have it act as a string replacement performed before handling the signs?

**Comment by Lee0701** [comment]
Thank you for pointing.
Will checking the server version before calling the method help?

**Comment by bergerkiller** [comment]
@Lee0701 Yeah, at least it won't error then. In BKCommonLib you can use Common.evaluateMCVersion for that like I pointed out in the comment. There may also be a field for precisely this in CommonCapabilities, but its internal code so I don't think you should use that.

**Comment by bergerkiller** [comment]
The most technical way is to use this method in PlayerUtil, and create a ResourceKey from the name, passing it into it:
https://github.com/bergerhealer/BKCommonLib/blob/master/src/main/java/com/bergerkiller/bukkit/common/utils/PlayerUtil.java#L380

**Comment by Lee0701** [comment]
This would catch NoSuchMethodError on lower versions.

**Comment by bergerkiller** [comment]
I'll allow it, it shifts responsibility for dealing with per-version logic to BKCommonLib where it belongs. I'll pull it and work on a more suitable fallback for this on 1.8.8 myself later today. :)

Similarly, I'll have to see what works best for longer sound names. Signs are limiting in general, would be nice to work around this interactively in some way. Not just for sounds.

**Comment by Lee0701** [comment]
Thank you! :D

**Comment by bergerkiller** [comment]
Looks like playSound with sound name is supported on 1.8 after all. I think I had it confused with a similar method using the Sound enumeration, which was changed radically at some version.

**Pull request merged**